File: lingting-spring-core/src/test/java/live/lingting/spring/SpringTestBean.java
package live.lingting.spring;

import live.lingting.framework.context.ContextComponent;
import lombok.Getter;
import org.springframework.stereotype.Component;

/**
 * @author lingting 2024-02-02 15:27
 */
@Getter
@Component
public class SpringTestBean implements ContextComponent {

	private final String value = "bean";

	private boolean start = false;

	@Override
	public void onApplicationStart() {
		start = true;
	}

	@Override
	public void onApplicationStop() {
		//
	}

}

File: lingting-spring-core/src/test/kotlin/live/lingting/spring/SpringTestBean.kt
package live.lingting.spring

import live.lingting.framework.application.ApplicationComponent
import org.springframework.stereotype.Component

/**
 * @author lingting 2024-02-02 15:27
 */
@Component
class SpringTestBean : ApplicationComponent {
    val value: String = "bean"

    var isStart: Boolean = false
        private set

    override fun onApplicationStart() {
        this.isStart = true
    }

    override fun onApplicationStop() {
        //
    }
}
